File: TP1/CouleurObjC/CouleurObjC/ViewController.m
//
//  ViewController.m
//  CouleurObjC
//
//  Created by m2sar on 19/09/2015.
//
//  @author Ilyas  Toumlilt - <ilyas.toumlilt@etu.upmc.fr>
//  @author Adrien Ferreira - <adrien.ferreira@etu.upmc.fr>
//
//  Copyright (c) 2015 m2sar. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

// La palette de couleurs :
NSArray* myColors;

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    myColors = [NSArray arrayWithObjects:
                [UIColor grayColor],
                [UIColor orangeColor],
                [UIColor redColor],
                [UIColor blueColor],
                [UIColor greenColor],
                [UIColor blackColor],
                nil];
    [_myStepper setMinimumValue:0.0];
    [_myColorView setBackgroundColor:myColors[0]];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)onChangeMyStepper:(id)sender {
    [_myColorView setBackgroundColor:myColors[(int)_myStepper.value]];
}
@end

File: TP3/QuizzSwift/QuizzSwift/ViewController.swift
//
//  ViewController.swift
//  QuizzSwift
//
//  Created by m2sar on 01/10/2015.
//  Copyright (c) 2015 m2sar. All rights reserved.
//

import UIKit


class ViewController: UIViewController {

    var QR = QuestionsReponses();
    
    @IBOutlet weak var myQuestionTV: UITextView!
    @IBOutlet weak var myAnswerTV: UITextView!
    @IBOutlet weak var myRspCounterLabel: UILabel!
    @IBOutlet weak var myBalaiseSwitch: UISwitch!
    @IBOutlet weak var myNextButton: UIButton!
    @IBOutlet weak var myPreviousButton: UIButton!
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        self.myAnswerTV.textColor = UIColor.greenColor();
        self.onChangeBalaiseSwitch(self.myBalaiseSwitch);
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func onClickReponse(sender: UIButton) {
        self.myAnswerTV.text = self.QR.getAnswer();
        self.myRspCounterLabel.text = "Réponses vues : \(self.QR.getRspCounter())";
    }

    @IBAction func onClickNext(sender: UIButton) {
        self.myNextButton.setImage(UIImage(named: "flecheDroite"), forState: .Normal);
        var lvl: Int;
        (lvl, self.myQuestionTV.text, self.myAnswerTV.text) = self.QR.getNextQuestion((self.myBalaiseSwitch.on ? HARD_LEVEL : EASY_LEVEL));
        self.myQuestionTV.textColor = ( lvl == HARD_LEVEL ) ? UIColor.redColor() : UIColor.purpleColor();
    }
    
    
    @IBAction func onTouchDownNextOrPrevious(sender: UIButton) {
        sender.setImage(UIImage(named: "fleche-en-action"), forState: .Normal);
    }
    
    @IBAction func onClickPrevious(sender: UIButton) {
        self.myPreviousButton.setImage(UIImage(named: "flecheGauche"), forState: .Normal);
        var lvl: Int;
        (lvl, self.myQuestionTV.text, self.myAnswerTV.text) = self.QR.getPreviousQuestion((self.myBalaiseSwitch.on ? HARD_LEVEL : EASY_LEVEL));
        self.myQuestionTV.textColor = ( lvl == HARD_LEVEL ) ? UIColor.redColor() : UIColor.purpleColor();
    }
    
    @IBAction func onChangeBalaiseSwitch(sender: UISwitch) {
        if( self.myBalaiseSwitch.on ) {
            self.onClickNext(self.myNextButton);
        } else {
            self.onClickPrevious(self.myPreviousButton);
        }
    }
}
